document/js-api/index.bs

```diff
@@ -1329,6 +1383,133 @@ The algorithm <dfn>ToWebAssemblyValue</dfn>(|v|, |type|) coerces a JavaScript va
+        1. Let |realm| be |func|'s [=associated Realm=].
+        1. Let |relevant settings| be |realm|'s [=realm/settings object=].
+        1. Let |async_context| be the [=surrounding agent=]'s [=running execution context=].
+        1. Let |map| be the [=surrounding agent=]'s associated [=Execution Context Status map=].
+        1. If the entry for |async_context| in |map| is not [=active=] then trap.
+        1. [=Prepare to run script=] with |relevant settings|.
+        1. [=Prepare to run a callback=] with |stored settings|.
+        1. Let [|parameters|] → [|resultTypes|] be |functype|.
+        1. Let |jsArguments| be the result of [=coerce WebAssembly arguments=](|parameters|,|arguments|).
+        1. Let |ret| be [=?=] [$Call$](|func|, undefined, |jsArguments|).
+        1. [=Clean up after running a callback=] with |stored settings|.
+        1. [=Clean up after running script=] with |relevant settings|.
+        1. Assert: |ret|.\[[Type]] is <emu-const>throw</emu-const> or <emu-const>normal</emu-const>.
+        1. If |ret|.\[[Type]] is <emu-const>throw</emu-const>, then:
+            1. Let |type|, |payload| and |opaqueData| be the result of [=coerce a JavaScript exception|coercing the JavaScript exception=] |ret|.\[[Value]].
+            1. [=WebAssembly/Throw=] with |type|, |payload| and |opaqueData|.
+        1. Otherwise, if [=list/size=] of |ret| is 1 and [$IsPromise$](|ret|.\[[Value]][0]):
+            1. Let |promise| be |ret|.\[[Value]][0].
+            1. Set the entry for |async_context| in |map| to [=paused=].
+            1. Let |awaitResult| be the result of performing [$Await$](|promise|).
+            1. Note: this will suspend both this algorithm, and the WebAssembly function being invoked by the [=evaluate a Promising function=] algorithm. On return, |ret| will be either a normal completion or an exceptional completion.
+            1. If the entry for |async_context| in |map| is not [=paused=] then trap, otherwise, set the entry to [=active=].
+            1. If |awaitResult|.\[[Type]] is <emu-const>throw</emu-const>, then:
+               1. Let |type|, |payload| and |opaqueData| be the result of [=coerce a JavaScript exception|coercing the JavaScript exception=] |ret|.\[[Value]].
+               1. [=WebAssembly/Throw=] with |type|, |payload| and |opaqueData|.
+            1. Otherwise, return the result of performing [=coerce a JavaScript return=] on |resultTypes| and |awaitResult|.
+        1. Otherwise, return the result of performing [=coerce a JavaScript return=] on |resultTypes| and |ret|.
+    1. Let |store| be the [=surrounding agent=]'s [=associated store=].
+    1. Let (|store|, |funcaddr|) be [=func_alloc=](|store|, |functype|, |hostfunc|).
+    1. Set the [=surrounding agent=]'s [=associated store=] to |store|.
+    1. Return |funcaddr|.
+
+Note that we only invoke [$Await$] on the result of calling the JavaScript function |func| if it has returned a Promise object.
```

Comment:
I don't know Bikeshed spec norms, but in 262 this note would be better as an inline `NOTE:` step next to the step it's a note for, instead of at the end.

Reply:
I put it inline.